core/vtk/ttkScalarFieldCriticalPoints/ttkScalarFieldCriticalPoints.cpp

```diff
@@ -78,6 +82,17 @@ int ttkScalarFieldCriticalPoints::RequestData(
   this->preconditionTriangulation(triangulation);
   this->setOutput(&criticalPoints_);
 
+#if TTK_ENABLE_MPI_TIME
+  Timer t_mpi;
+  // Get processes information
+  vtkMPIController *controller = vtkMPIController::SafeDownCast(
+    vtkMultiProcessController::GetGlobalController());
```

Comment:
Can we do differently than using VTK's handles to the MPI's controller?
Isn't it possible to access MPI communications from the base code without any interaction with VTK's handles?

Reply:
If we assume to use the default communicator (MPI_COMM_WORLD) for TTK, which should always be available, we can use MPI_BARRIER(MPI_COMM_WORLD), without using the VTK functions. 

In the base layer we _need_ to use 'raw' MPI because one could use TTK standalone without VTK, in the vtk layer it should be no problem to rely on the VTK handles, however I would also use the raw MPI for consistency.

Reply:
that was the point of my question indeed.
I'd recommend to implement the MPI timing util functionalities at the base code level indeed.